### PR TITLE
[Tf 2725] Fix and Improve Fingerprinting

### PR DESCRIPTION
**Pull request by TinyGambe**
# [Tf 2725] Fix and Improve Fingerprinting
This PR fixes scripts that were not able to run because some methods / variables were missing. This also improves fingerprinting by making the fingerprinting more consistent and sending the correct user agent on http headers.

## Modifying Http Requests
Altering the client side User Agent (UA) and other details is insufficient as the UA being sent in the HTTP headers still has "headless" etc. This PR also modifies the Http Headers so that UA does not have headless 

## For the future
Note that the fingerprinting is temporarily hardcoded but in the future, we can randomize this browser fingerprinting but we need to group different browser fingerprints together so that it is consistent (a platform with MacOS shouldn't have a Windows browser fingerprinting). 

## Tests
Successfull tested with reddit.com, bestbuy.com, tesla.com.
Screenshot 2024-10-10 at 4 43 05 PM
Screenshot 2024-10-10 at 4 43 46 PM

Screenshot 2024-10-10 at 4 44 09 PM

Partial success on walmart.com, etsy.com (Means that sometimes, it fails)
Screenshot 2024-10-10 at 3 54 11 PM (2)
Screenshot 2024-10-10 at 3 02 45 PM (2)

## Bot Tests (Headless)
example_with_stealth

### CODE REVIEW

File: playwright_stealth/stealth.py [hunk 1]
```diff
@@ -149,6 +150,35 @@ async def stealth_async(page: AsyncPage, config: StealthConfig = None):
...
+
+
+async def add_stealth_headers(page: AsyncPage):
+    """Adds stealth headers to the page"""
+    page.route(
+        "**/*",
+        lambda route, request: route.continue_(
+            headers={
+                **request.headers,
+                "sec-ch-ua": '"Chromium";v="129", "Not=A?Brand";v="8"',
```
Comment (lozzle) [inline playwright_stealth/stealth.py:167]: curious what the rationale is for setting a full explicit string vs removing keys via regex?  iiuc if we lock our fingerprint in to something very specific as is here, this can also over time be an indication to sites that this exact fingerprint is a bot, vs leveraging existing headers and only modifying the giveaway signals means there will still be variance

File: playwright_stealth/js/chrome.app.js [hunk 2]
```diff
@@ -16,60 +16,62 @@ window.chrome = {
...
-  "READY_TO_RUN": "ready_to_run",
-  "RUNNING": "running"
-  }
-  }
-      `.trim()
-);
-
-window.chrome.app = {
-  ...STATIC_DATA,
+  // return; // Nothing to do here
```
Comment (paveldudka) [inline playwright_stealth/js/chrome.app.js:19]: Whats up with these commented out returns? We cant just comment out return. Essentially by doing this we disable "if" check. If thats our intention - instead of commenting out return, that check needs to be removed/reworked. And it would be helpful to provide rationale for such rework as I bet those checks are there for a reason.

Honestly, this PR is massive to be reviewed in one go. Very hard to parse it.
Is there a way to break it down?

**Review by paveldudka — CHANGES_REQUESTED** [review]

File: playwright_stealth/js/chrome.app.js (hunk 2 above)
Comment (TinyGambe) [reply playwright_stealth/js/chrome.app.js:19]: I forgot to remove these commented `return` statements. Initially, these return statements causes the rest of the script to not run. So, I wrapped it in an if else statement instead. I'll try to split the PR tomorrow but it's quite hard to because the scripts are linked together.

File: playwright_stealth/stealth.py (hunk 1 above)
Comment (TinyGambe) [reply playwright_stealth/stealth.py:167]: I'm still experimenting on this but there is still some header & client side inconsistencies for some reason. I need more time on researching which combinations and attributes will result in greater consistency.

File: playwright_stealth/js/chrome.app.js [hunk 3]
```diff
@@ -15,61 +15,61 @@ window.chrome = {
...
+    "isInstalled": false,
+    "InstallState": {
+    "DISABLED": "disabled",
+    "INSTALLED": "installed",
+    "NOT_INSTALLED": "not_installed"
+    },
+    "RunningState": {
+    "CANNOT_RUN": "cannot_run",
+    "READY_TO_RUN": "ready_to_run",
+    "RUNNING": "running"
```
Comment (colriot) [inline playwright_stealth/js/chrome.app.js:41]: not? let's apply some formatting/indentation here for readability?

File: playwright_stealth/js/navigator.userAgent.js [hunk 4]
```diff
@@ -91,16 +96,18 @@ const _getDeviceMemory = () => {
...
-    architecture: _getPlatformArch(),
-    model: _getPlatformModel(),
-    mobile: _getMobile(),
+  userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36",
+  platform: "macOS",
+  userAgentData: {
+    brands: [
+      {
+        brand: "Chromium",
+        version: "129",
```
Comment (colriot) [inline playwright_stealth/js/navigator.userAgent.js:105]: should we extract these values into constants to make sure we can update and configure them separately from wandering the source code?
